adapters/repos/db/vector/flat/index.go

```diff
@@ -1052,9 +1052,37 @@ func (index *flat) Stats() (common.IndexStats, error) {
 	return &FlatStats{}, errors.New("Stats() is not implemented for flat index")
 }
 
-func (index *flat) VectorStorageSize() int64 {
-	// TODO-usage: Implement this
-	return 0
+func (index *flat) VectorStorageSize(_ context.Context) int64 {
+	count := atomic.LoadUint64(&index.count)
```

Comment:
`index.count` in the flat index is again more of a hint for optimization vs the most accurate number. It would be best to use `CalcTargetVectorDimensionsFromStore`.

Additionally it may be worth caching `CalcTargetVectorDimensionsFromStore` or the metrics in general at some layer. Currently when shard dimensions tracking is enabled they run every (from memory 5) minutes and update the prometheus metrics then.

Reply:
pushed a change, we could do LRU cache or so but i'd prefer to do this mechanism in separate PR, wehere we intrdouce caching for vector dimensions calculations